### PR TITLE
Ignore unfulfillable jobs

### PR DESCRIPTION
**Pull request by FinchPowers**

### CODE REVIEW

File: starcluster/balancers/sge/__init__.py [hunk 1]
```diff
@@ -36,6 +37,11 @@
 DEFAULT_STATS_FILE = os.path.join(DEFAULT_STATS_DIR, 'sge-stats.csv')
 
 
+class SGEJob(dict):
+    def __hash__(self):
+        return int(self['JB_job_number'])
```
Comment (jraby) [inline starcluster/balancers/sge/__init__.py:42]: What should happen if there's nothing in the hash?
return None or raise?

File: starcluster/balancers/sge/__init__.py [hunk 2]
```diff
@@ -719,10 +739,23 @@ def _eval_required_instances(self):
                      "and no queued jobs...")
             return 0
 
+        job_instances_support = self.get_jobs_instances_support(queued_jobs)
+        for job in job_instances_support['unfulfillable']:
+            log.warning("Job %s cannot be fulfilled because it is requesting "
+                        "more resource than any of the configured node has.")
```
Comment (jraby) [inline starcluster/balancers/sge/__init__.py]: seem to be missing a `, job`  in the log call

File: starcluster/balancers/sge/__init__.py (hunk 1 above)
Comment (FinchPowers) [inline starcluster/balancers/sge/__init__.py:42]: It would be a bug in OGS or in the qstat command. There should always be a JB_job_number.

**Comment by FinchPowers** [comment]
Squash merged.

**Pull request closed without merging**